Add command-line instructions to install the hubble CLI (#12001)
cilium-team/cilium: Documentation/gettingstarted/hubble-install.rst

.. tabs::

   .. group-tab:: Linux

      Download the latest hubble release:

      .. parsed-literal::

>          curl -L https://github.com/cilium/hubble/releases/latest/download/hubble-linux-amd64.tar.gz | tar zxf -

Reviewer: ```suggestion
         curl -LO https://github.com/cilium/hubble/releases/latest/download/hubble-linux-amd64.tar.gz
         curl -LO https://github.com/cilium/hubble/releases/latest/download/hubble-linux-amd64.tar.gz.sha256sum
         sha256sum --check hubble-linux-amd64.tar.gz.sha256sum && tar zxf hubble-linux-amd64.tar.gz
```

Author: ok let's do it. i went back and forth on this one, but i agree it's worth it.

>          curl -L https://github.com/cilium/hubble/releases/latest/download/hubble-linux-amd64.tar.gz | tar zxf -

Reviewer 2: Don't you want to pin a specific version instead of using latest? (at least in the Cilium's stable branches)

Reviewer: I think this is done on purpose: https://github.com/cilium/hubble/issues/292

Author: @aanm @Rolinh i was a bit hesitant to pin a version in the cilium documentation. i felt like that would add too much overhead for maintaining the documentation, and it's easy to forget to keep it up to date.

how about having `stable.txt` like how kubectl does it: https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-linux this way the documentation doesn't have to be updated, and we can control which version is considered "stable" from hubble release page.

Author: that would probably be a separate PR though...

Reviewer 2: > @aanm @Rolinh i was a bit hesitant to pin a version in the cilium documentation. i felt like that would add too much overhead for maintaining the documentation, and it's easy to forget to keep it up to date.
> 
> how about having `stable.txt` like how kubectl does it: https://kubernetes.io/docs/tasks/tools/install-kubectl/#install-kubectl-on-linux this way the documentation doesn't have to be updated, and we can control which version is considered "stable" from hubble release page.

This is a great idea!

Author: https://github.com/cilium/hubble/issues/298

>       and move the ``hubble`` CLI to a directory listed in the ``$PATH`` environment variable:

Reviewer: ```suggestion
      and move the ``hubble`` CLI to a directory listed in the ``$PATH`` environment variable. For example:
```

      .. parsed-literal::

         sudo mv hubble /usr/local/bin

   .. group-tab:: MacOS

      Download the latest hubble release:

      .. parsed-literal::

         curl -L https://github.com/cilium/hubble/releases/latest/download/hubble-darwin-amd64.tar.gz | tar zxf -

>       and move the ``hubble`` CLI to a directory listed in the ``$PATH`` environment variable:

Reviewer: ```suggestion
      and move the ``hubble`` CLI to a directory listed in the ``$PATH`` environment variable. For example:
```

      .. parsed-literal::

         sudo mv hubble /usr/local/bin

   .. group-tab:: Windows

      Download the latest hubble release:

      .. parsed-literal::

         curl -L https://github.com/cilium/hubble/releases/latest/download/hubble-windows-amd64.tar.gz

>       and move the ``hubble.exe`` CLI to your ``$PATH`` after extracting it from the tarball.

Reviewer: ```suggestion
      and move the ``hubble.exe`` CLI to a directory listed in the ``%PATH%`` environment variable after extracting it from the tarball.
```
On Windows, it's `%PATH%`, not `$PATH`. Not sure if we can give an example with the default `%PATH%` as well as I don't know what the common practice is.

Author: there is only so much i can do man. let's change this to `%PATH%` and call it a day

      Download the latest hubble release:

      .. parsed-literal::

         curl -LO https://github.com/cilium/hubble/releases/latest/download/hubble-linux-amd64.tar.gz
         curl -LO https://github.com/cilium/hubble/releases/latest/download/hubble-linux-amd64.tar.gz.sha256sum
         sha256sum --check hubble-linux-amd64.tar.gz.sha256sum
         tar zxf hubble-linux-amd64.tar.gz

      and move the ``hubble`` CLI to a directory listed in the ``$PATH`` environment variable. For example:

      .. parsed-literal::

         sudo mv hubble /usr/local/bin

   .. group-tab:: MacOS

      Download the latest hubble release:

      .. parsed-literal::

         curl -LO https://github.com/cilium/hubble/releases/latest/download/hubble-darwin-amd64.tar.gz
         curl -LO https://github.com/cilium/hubble/releases/latest/download/hubble-darwin-amd64.tar.gz.sha256sum
         sha256sum --check hubble-darwin-amd64.tar.gz.sha256sum
         tar zxf hubble-darwin-amd64.tar.gz

      and move the ``hubble`` CLI to a directory listed in the ``$PATH`` environment variable. For example:

      .. parsed-literal::

         sudo mv hubble /usr/local/bin

   .. group-tab:: Windows

>       Download the latest hubble release:
> 
>       .. parsed-literal::
> 
>          curl -L https://github.com/cilium/hubble/releases/latest/download/hubble-windows-amd64.tar.gz
> 
>       and move the ``hubble.exe`` CLI to a directory listed in the ``%PATH%`` environment variable after
>       extracting it from the tarball.

Reviewer: I think the Windows instructions are lacking as the tarball is never extracted. By the way, is `curl` available on Windows? How does one download a file, verify the SHA256 checksum and extract a `.tar.gz` file natively on Windows?

EDIT: it looks like on recent versions of Windows, `curl.exe` and `tar` are [installed by default](https://techcommunity.microsoft.com/t5/containers/tar-and-curl-come-to-windows/ba-p/382409).

EDIT2: got access to a machine with Windows 10. It seems that the following instructions work with native tools. I'm still not sure what a recommended directory in `%PATH%` would be.

```suggestion
      Download the latest hubble release:

      .. parsed-literal::

         curl -LO https://github.com/cilium/hubble/releases/latest/download/hubble-windows-amd64.tar.gz
         curl -LO https://github.com/cilium/hubble/releases/latest/download/hubble-windows-amd64.tar.gz.sha256
         certutil -hashfile hubble-windows-amd64.tar.gz SHA256
         type hubble-windows-amd64.tar.gz.sha256
         # verify that the checksum from the two commands above match
         tar zxf hubble-windows-amd64.tar.gz

      and move the ``hubble.exe`` CLI to a directory listed in the ``%PATH%`` environment variable after
      extracting it from the tarball.
```

Author: thanks @Rolinh this is super-helpful. here is the updated doc: https://5ee2681c83fec8187682a6e9--cilium-docs.netlify.app/gettingstarted/k8s-install-gke.html#enable-hubble
